libevmasm/AssemblyItem.cpp

```diff
@@ -243,15 +245,15 @@ string AssemblyItem::getJumpTypeAsString() const
 	}
 }
 
-string AssemblyItem::toAssemblyText(Assembly const& _assembly) const
+string AssemblyItem::toAssemblyText(Assembly const& _assembly, langutil::EVMVersion _evmVersion) const
```

Comment:
Why not use ``_assembly.evmVersion()`` in here? We should probably do either that, or actually remove the evm-version from ``Assembly``. It should be possible to store it in ``Functionalizer`` instead, which would mean making it an argument of ``Assembly::assemblyStream`` and then to ``Assembly::assemblyString``, but all the call sites of that should have the evm version available, right? Might be that I'm missing another reason for having it in ``Assembly``.

Reply:
Indeed. I passed `evmVersion` to `toAssemblyText` before I added the method in the `Assembly` class and forgot to get back to it. I will fix it.
Yes, making it an argument of `assemblyStream` and `assemblyString` would require pass it in a couple of more places, which indeed have the evm version available in the caller. However, it would also require that we pass the version to `operator<<` since it uses `assemblyStream`.

Reply:
Yeah, I also just realized that having it in ``Assembly`` will come in handy later for implementing EOF EIP-663 to determine how many swaps and dups we have available, so good to keep it there!